### PR TITLE
Shedule tasks from config #8719

### PR DESCRIPTION
**Pull request by vbradnitski**

### CODE REVIEW

File: modules/core/core-scheduler/src/main/java/com/enonic/xp/impl/scheduler/SchedulerConfigImpl.java [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+import com.enonic.xp.scheduler.CreateScheduledJobParams;
+import com.enonic.xp.scheduler.ScheduleCalendarType;
+import com.enonic.xp.scheduler.SchedulerName;
+import com.enonic.xp.security.PrincipalKey;
+
+@Component(configurationPid = "com.enonic.xp.scheduler")
+public class SchedulerConfigImpl
+    implements SchedulerConfig
+{
+    private static final Pattern PATTERN = Pattern.compile( "^([a-z0-9]+)\\.([a-z0-9]+)(?:\\.([a-z0-9]+))?$" );
```
Comment (rymsha) [inline modules/core/core-scheduler/src/main/java/com/enonic/xp/impl/scheduler/SchedulerConfigImpl.java:34]: Give nice name to this pattern
Also, give names to matching groups

File: modules/core/core-scheduler/src/main/java/com/enonic/xp/impl/scheduler/SchedulerConfigImpl.java [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+        final Map<SchedulerName, ScheduleCalendarType> calendarTypes = new HashMap<>();
+        final Map<SchedulerName, TimeZone> calendarTimezones = new HashMap<>();
+        final Map<SchedulerName, String> calendarValues = new HashMap<>();
+
+        for ( final Map.Entry<String, String> entry : jobConfig.asMap().entrySet() )
+        {
+            final String value = entry.getValue();
+
+            final Matcher matcher = PATTERN.matcher( entry.getKey() );
+            if ( matcher.find() )
```
Comment (rymsha) [inline modules/core/core-scheduler/src/main/java/com/enonic/xp/impl/scheduler/SchedulerConfigImpl.java:77]: Single find is better be replaced with match (also change pattern a bit)

File: modules/core/core-scheduler/src/main/java/com/enonic/xp/impl/scheduler/SchedulerConfigImpl.java [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+            {
+                final String jobName = matcher.group( 1 );
+                final String propertyName = matcher.group( 2 );
+
+                final SchedulerName schedulerName = SchedulerName.from( jobName );
+
+                jobs.putIfAbsent( schedulerName, CreateScheduledJobParams.create().name( schedulerName ) );
```
Comment (rymsha) [inline modules/core/core-scheduler/src/main/java/com/enonic/xp/impl/scheduler/SchedulerConfigImpl.java:84]: Maybe throw exception, if duplicate jobs found?

File: modules/core/core-scheduler/src/main/java/com/enonic/xp/impl/scheduler/SchedulerServiceActivator.java [hunk 4]
```diff
@@ -60,6 +71,36 @@ public void activate( final BundleContext context )
...
+    private void createConfigJobs( final SchedulerService schedulerService )
+    {
+        adminContext().runWith( () -> schedulerConfig.jobs().forEach( job -> {
+            try
+            {
+                schedulerService.create( job );
+            }
+            catch ( Exception e )
+            {
+                LOG.debug( String.format( "Can't create [%s] job.", job.getName().getValue() ) );
```
Comment (rymsha) [inline modules/core/core-scheduler/src/main/java/com/enonic/xp/impl/scheduler/SchedulerServiceActivator.java:88]: This is too broad to swallow.
Duplicate job exception should be swallowed.
Others should be well visible

**Review by rymsha — APPROVED** [review]

File: modules/core/core-scheduler/src/main/java/com/enonic/xp/impl/scheduler/SchedulerConfigImpl.java [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+import com.enonic.xp.form.PropertyTreeMarshallerService;
+import com.enonic.xp.page.DescriptorKey;
+import com.enonic.xp.scheduler.CalendarService;
+import com.enonic.xp.scheduler.CreateScheduledJobParams;
+import com.enonic.xp.scheduler.ScheduleCalendar;
+import com.enonic.xp.scheduler.ScheduleCalendarType;
+import com.enonic.xp.scheduler.SchedulerName;
+import com.enonic.xp.security.PrincipalKey;
+
+@Component(configurationPid = "com.enonic.xp.scheduler")
```
Comment (anatol-sialitski) [inline modules/core/core-scheduler/src/main/java/com/enonic/xp/impl/scheduler/SchedulerConfigImpl.java:31]: @vbradnitski missing the `com.enonic.xp.scheduler.cfg` file in the `https://github.com/enonic/xp/tree/epic-scheduler/modules/runtime/src/home/config` folder. Please add it.

Comment (vbradnitski) [reply modules/core/core-scheduler/src/main/java/com/enonic/xp/impl/scheduler/SchedulerConfigImpl.java:31]: it's not missing, it's there

Comment (anatol-sialitski) [reply modules/core/core-scheduler/src/main/java/com/enonic/xp/impl/scheduler/SchedulerConfigImpl.java:31]: Yes, I can see now. Please, do not forget to point which properties should be used for a job configuration.

**Review by anatol-sialitski — APPROVED** [review]

**Pull request merged**